### ISSUE TITLE
Delay before expand (or forbid to) expand node

### ISSUE THREAD

**Issue by oarashi** [fritz-c/react-sortable-tree#141]
Is it possible to delay (or forbid to) the expanding of element when dragging another element over it?
I want to drag one element (witch is located at the top of the tree) to the bottom. And while I'm doing it all components became expanded. And I want them to stay collapsed.
I've tried throttle and debounce but it didn't help.
Thanks!
P.S. Great library! =)

**Comment by wuweiweiwu** [comment]
@oarashi I would have to double check but if you set canDrop to false for the top level nodes then they should not expand. I would also consider adding a `expandable` prop. Feel free to open a PR :)

**Comment by danblaker** [comment]
@wuweiweiwu I agree with @oarashi that this is a terrific library; however, the current instant-expand behavior severely limits the usability of the interface. I've watched a number of users struggle with trying to find the destination container as nodes expand while they drag; in most cases, their target moves unexpectedly when their drag cursor approaches it because the node before it has expanded. Rather than a prop to *prevent* expansion, we need a prop to *delay* expansion.

Here are some articles discussing how important it is to delay showing/hiding items on hover:
https://www.nngroup.com/articles/timing-exposing-content/
https://www.technologyreview.com/s/512376/the-ingenious-engineering-trick-that-makes-amazon-menus-usable/
https://css-tricks.com/dropdown-menus-with-more-forgiving-mouse-movement-paths/

**Comment by buzztone** [comment]
I think there is definitely a need to forbid expansion of collapsed nodes in some use cases.

We use [Drag from external source](https://fritz-c.github.io/react-sortable-tree/storybook/?selectedKind=Advanced&selectedStory=Drag%20from%20external%20source&full=0&addons=0&stories=1&panelRight=0) to add nodes to an existing tree. 

Our users often have significant problems adding new nodes when existing collapsed nodes expand to full the UI. 

In our view users have chosen to collapse nodes for a reason and this expansion is unexpected and very disruptive.

**Comment by dukoo** [comment]
I'm wondering if we can expect to have some answers from the authors?:)

**Comment by cacabo** [comment]
Yeah agree there should definitely be an expand on drag delay offset prop. Especially if this library's goal is to mimic the way popular tools like Notion handle these interactions.
